gateway/gateway.go

```diff
@@ -232,19 +237,25 @@ func (gateway *HandleT) webRequestBatchDBWriter(process int) {
 			//Should be function of body
 			newJob := jobsdb.JobT{
 				UUID:         id,
-				Parameters:   []byte(fmt.Sprintf(`{"source_id": "%v"}`, enabledWriteKeysSourceMap[writeKey])),
+				UserID:       gjson.GetBytes(body, "batch.0.anonymousId").Str,
+				Parameters:   []byte(fmt.Sprintf(`{"source_id": "%v", "batch_id": %d}`, enabledWriteKeysSourceMap[writeKey], counter)),
 				CustomVal:    CustomVal,
 				EventPayload: []byte(body),
 			}
 			jobList = append(jobList, &newJob)
+
+			if computedNode := gateway.pf.GetNodeFromUserID(newJob.UserID); computedNode.GetNodeID() != misc.GetNodeID() {
```

Comment:
add an if condition to check only if migrator feature available ? 
We might not want to do these checks for some deplyments (where we don't want multi node features?) even it's an enterprise build and migratior feature is available. Should we also have a flag for that - application.Features().Migrator.Enabled() ?

Reply:
This piece must go away before we merge.